packages/turbo-workspaces/src/updateDependencies.ts

```diff
@@ -0,0 +1,127 @@
+import fs from "fs-extra";
+import chalk from "chalk";
+import path from "path";
+import {
+  Project,
+  Workspace,
+  DependencyList,
+  PackageManagerDetails,
+  Options,
+  PackageJsonDependencies,
+} from "./types";
+import { Logger } from "./logger";
+import { getPackageJson } from "./utils";
+
+function updateDependencyList({
+  dependencyList,
+  project,
+  to,
+}: {
+  dependencyList: DependencyList;
+  project: Project;
+  to: PackageManagerDetails;
+}): { dependencyList: DependencyList; updated: Array<string> } {
+  const updated: Array<string> = [];
+  project.workspaceData.workspaces.forEach((workspace) => {
+    const { name } = workspace;
+    if (dependencyList[name]) {
+      const workspaceVersion = dependencyList[name];
+      const version = workspaceVersion.startsWith("workspace:")
+        ? workspaceVersion.slice("workspace:".length)
+        : workspaceVersion;
+      dependencyList[name] =
+        to.name === "pnpm" ? `workspace:${version}` : version;
+      updated.push(name);
+    }
+  });
+
+  return { dependencyList, updated };
+}
+
+export default function updateDependencies({
+  project,
+  workspace,
+  to,
+  logger,
+  options,
+}: {
+  workspace: Workspace;
+  project: Project;
+  to: PackageManagerDetails;
+  logger: Logger;
+  options?: Options;
+}) {
+  // update all dependencies
+  const workspacePackageJson = getPackageJson({
+    workspaceRoot: workspace.paths.root,
+  });
+
+  // collect stats as we go for consolidated output at the end
+  const stats: Record<keyof PackageJsonDependencies, Array<string>> = {
+    dependencies: [],
+    devDependencies: [],
+    peerDependencies: [],
+    optionalDependencies: [],
+  };
+
+  const allDependencyKeys: Array<keyof PackageJsonDependencies> = [
+    "dependencies",
+    "devDependencies",
+    "peerDependencies",
+    "optionalDependencies",
+  ];
+
+  allDependencyKeys.forEach((depKey) => {
+    const depList = workspacePackageJson[depKey];
+    if (depList) {
+      const { updated, dependencyList } = updateDependencyList({
+        dependencyList: depList,
+        project,
+        to,
+      });
+
+      workspacePackageJson[depKey] = dependencyList;
+      stats[depKey] = updated;
+    }
+  });
+
+  const toLog = (key: keyof PackageJsonDependencies) => {
+    const total = stats[key].length;
+    if (total > 0) {
+      return `${chalk.green(total.toString())} ${key}`;
+    }
+    return undefined;
+  };
+
+  const allChanges = allDependencyKeys.map(toLog).filter(Boolean);
+  const workspaceLocation = `./${path.relative(
```

Comment:
```suggestion
  const workspaceLocation = `${path.relative(
```

We allow passing `dir` so it isn't `./`.

Reply:
it is, I'm explicitly making this relative to root to avoid really long paths in the output.